nucliadb_vectors2/src/vectors/indexset/state.rs

```diff
@@ -50,7 +50,15 @@ impl State {
         }
         Ok(())
     }
-    pub fn new_index(&mut self, index: String) -> VectorR<Index> {
+    pub fn get(&self, index: &str) -> VectorR<Option<Index>> {
+        if self.indexes.contains(index) {
+            let location = self.location.join(index);
+            Some(Index::new(&location, false)).transpose()
+        } else {
+            Ok(None)
+        }
+    }
+    pub fn get_or_create(&mut self, index: String) -> VectorR<Index> {
```

Comment:
We can save a `clone` call when the index already exists by using `index: impl Into<Cow<'static, str>>` and call `to_string` later on.

Reply:
will do